### DISCUSSION TITLE
Tailwind-inspirerte responsive klasser

### DISCUSSION THREAD

**Discussion by joms** [fremtind/jokul#2276] [Annet]
Hvilke steg i [prosessen](https://fremtind.github.io/jokul/komigang/prosessen) er utført?
- [ ] Undersøke
- [ ] Rapportere
- [ ] Avklare
- [ ] Spesifisere krav
- [ ] Utvikle
- [ ] Ferdig!

**Er ønsket tilknyttet et problem? Fortell oss mer!**
Jeg er veldig glad i å bruke klasser som `jkl-component-spacing--large-left` og lignende. Men de mister fort sin verdi når man skal begynne med responsivt design hvor spacing, teksttyper og lignende kan endre seg.

**Beskriv løsningen du ønsker**
I TailwindCSS genererer man en syntax hvor man kan prefixe klasser med f.eks. `mobile:` for å spesifisere at denne klassen kun skal gjelde innenfor breakpointet til `mobile`: https://tailwindcss.com/docs/responsive-design

Hvis vi skal fortsette å satse på å eksponere klasser synes jeg dette eller en lignende løsning er noe vi bør ha.

Tanker @piofinn, @Saegrov, @lfbergee, @vegardsoerlie, @Mikaila94?

**Comment by lfbergee** [comment]
Tommel opp for utilityklasser!

**Comment by piofinn** [comment]
Liker konseptet veldig godt for ting som spacing!

Tror vi skal være forsiktige med å gå _for_ langt i retning av Tailwind ellers, så vi ikke ender opp med å "viske ut" betydningen av komponenter. Men for ting som støtter opp under tingene _utenfor_ komponentene, som f.eks. layout og spacing, synes jeg dette er en god idé!

Angående navngiving tenker jeg vi bør prøve å kvitte oss med mobil/desktop-dikotomien litt. Det mest naturlige hadde vel vært å knytte prefiksene opp mot breakpoint'ene våre (`small-device`, `medium-device`, etc.). Det er kanskje ogsåp en god anledning til å se nærmere på de breakpointene? i dag får man f.eks. samme layout på de fleste mobiler i stående og liggende format, selv om det gir to diametralt motsatte størrelsesforhold på skjermen.
